tests/ui/static/reference-to-mut-static.e2024.stderr

```diff
@@ -35,7 +35,7 @@ LL |         let ref _a = addr_of!(X);
    |                      ~~~~~~~~~~~
```

Comment:
This suggestion is actually wrong too. Is this handled via the `stmt` path?

Reply:
handled via `maybe_stmt_static_mut`. Fixed.